application-accelerator/creating-accelerators/testing-accelerators.hbs.md

```diff
@@ -0,0 +1,134 @@
+
+## <a id="creating-accelerators-ci-cd-pipeline"></a>CI/CD Pipeline
+As you iterate on an accelerator, you may wish to have some automated assertions that run before any changes to the accelerator are accepted.
+
+With the `tanzu accelerator generate-from-local` command you can run a modified accelerator,
+specify a set of options and run assertions on the generated project.
+
+You can run the accelerator using the components on your TAP cluster, without impacting the state of the TAP cluster.
+
+To do so ensure you have the following prerequisites in your CI/CD environment:
+
+- The Tanzu CLI is available, with the Application Accelerator plugin. For suggestions, see [Getting the Tanzu CLI in a CI/CD pipeline](#tanzu-cli-in-ci-cd)
+- The server URL is pointing to the TAP cluster you want to test with. For details on setting the server URL, see [Application Accelerator CLI plug-in overview](../../cli-plugins/accelerator/overview.md)
+
+> *Note:* No changes will be made to the TAP cluster that is provided via the server URL.
+> No new accelerators/fragments will be registered or modified. A TAP cluster is
+> required in order to ensure consistency between the version that is used for testing
+> and the version that is used when the accelerator is registered. Furthermore, it allows
+> using registered fragments and accelerators as dependencies for the local accelerator/fragment.
+
+To test the accelerator that is located at the path `workspace/java-rest` run the following command:
+
+```console
+tanzu accelerator generate-from-local --accelerator-path java-rest=workspace/java-rest --fragment-names tap-workload,java-version --options '{"projectName":"test"}' --output-dir generated-project
+```
+
+This will generate the project in the local directory `generated-project`, using the 
+accelerator located at `workspace/java-rest` on the CI/CD system, the fragments 
+`tap-workload` and `java-version` which are assumed to be already registered in the TAP 
+cluster and the option `projectName` set to `test`.
+
+For the full documentation for the `generate-from-local` command see the reference [tanzu accelerator generate-from-local](../../cli-plugins/accelerator/command-reference/tanzu_accelerator_generate-from-local.md)
+
+Now that the generated project is available, you can run various assertions on it:
+```console
+cd generated-project
+test -f build.gradle
+./gradlew test
+```
+
+If you have multiple assertions, you may choose to run a pre-defined script:
+```console
+cd generated-project
+./run-assertions.sh
+```
+
+You may choose to generate multiple projects from the same accelerator, providing different
+options for each and running different assertions on each generated project.
+
+### <a id="tanzu-cli-in-ci-cd"></a>Getting the Tanzu CLI in a CI/CD pipeline
```

Comment:
Is this really something we recommend? In particular it skews download numbers artificially IMO

Reply:
I was also thinking about this, but here are the reasons I chose to recommend this approach:
1) I doubt that outside users will be allowed to include the CLI executable in their source code (especially if the accelerator is public) because the CLI is commercial software. 
2) Assuming they are always using the same token, it will skew download numbers, but not unique downloads, since all CI/CD downloads will be attributed to the same user.